src/core/stdcpp/utility.d

```diff
@@ -0,0 +1,35 @@
+/**
+ * D header file for interaction with Microsoft C++ <utility>
+ *
+ * Copyright: Copyright (c) 2018 D Language Foundation
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Manu Evans
+ * Source:    $(DRUNTIMESRC core/stdcpp/utility.d)
+ */
+
+module core.stdcpp.utility;
+
+import core.stdcpp.xutility : StdNamespace;
+
+extern(C++, (StdNamespace)):
+@nogc:
+
+/**
+* D language counterpart to C++ std::pair.
+*
+* C++ reference: $(LINK2 https://en.cppreference.com/w/cpp/utility/pair)
+*/
+struct pair(T1, T2)
+{
+    ///
+    alias first_type = T1;
+    ///
+    alias second_type = T2;
+
+    ///
+    T1 first;
+    ///
+    T2 second;
+}
```

Comment:
The freebsd failure is real unfortunately. `pair` is define here as a POD while in C++ it isn't. A copy constructor definition here should fix it.

Reply:
It seems `std::pair` is a POD by default on every platform except on freebsd.
https://github.com/llvm-mirror/libcxx/blob/41e7283b76a113234f40b72ab3b92644c0f04613/include/__config#L116-L118
https://github.com/llvm-mirror/libcxx/blob/8ef0e6a4fe63c27be4bddf80b8932133e102ba17/include/utility#L286-L301

Reply:
Adding an explicit constructor would mean that brace-style struct initializers for `pair` would work on all platforms other than FreeBSD. As an alternative, how about adding a trivial destructor on FreeBSD? I'll try pushing that and see if the tests then pass.